cisco_ise/actions/lib/base.py

```diff
@@ -0,0 +1,458 @@
+
+        # Most results will return useful data to intepret results.
+        # Creates and Deletes only return details when something goes wrong.
+        # Look for quiet success codes, and grab a useful message to send from
+        # HTTP_STATUS_RESPONSE_MESSAGE
+        response_msg = ''
+        for status, msg in HTTP_STATUS_RESPONSE_MESSAGE:
+            if response_code == status:
+                response_msg = msg
+                break
+
+        # Many fauilure conditions will result in the action succeeding.
+        # This isn't a bad thing, but we need to have a simple check to know
+        # If what we tried worked or not. Any non 2xx reponse means something didn't work
+        # https://developer.cisco.com/docs/identity-services-engine/#!error-codes
+        if client.ok:
+            succeeded = True
+        else:
+            succeeded = False
+
+        return (response_data, response_code, succeeded, response_msg)
+
+    def retieve_config_connection(self, connection_name=None):
+        """ Checks the pack config for a named connection entry
+            and returns a dict with connection params from config entry.
+            If no config found, returns an empty dict {}
+        """
+
+        try:
+            config_connection = self.config['connections'].get(
+                connection_name, None)
+        except Exception as err:
+            self.logger.error(f'Specified connection "{connection_name}" Error: {err}')
+            sys.exit(1)
+        # If we found a valid config (not None), set it as the connection
+        if config_connection:
+            self.logger.debug(
+                f'Connection with name "{connection_name}" loaded from pack configuration')
+            return config_connection
+        # didn't find the connection_name in the config
+        else:
+            self.logger.warning(
+                f'Connection with name "{connection_name}" not found in configuration')
+            # return an empty dictionary
+            connection = {}
+            return connection
+
+    def format_action_result(self,
+                             response_data=None, response_code=None,
+                             succeeded=None, response_msg=None):
+        """ Handles formatting to ensure consistent results formatting
+        """
+        return {'data': response_data,
+                'response_code': response_code,
+                'succeeded': succeeded,
+                'response_msg': response_msg}
+
+    def _setup_connection(self, connection_fields):
+        """ Handles the logic for determining what credentials/hostname to use.
+            Resolves conflicts between config values, and action values
+
+            Priority: Action Specified Value > Config Value
+        """
+
+        try:
+            self.action_kwargs
+        except NameError:
+            raise NameError(
+                '_prepare_kwargs() is required to be ran prior to _setup_connection()')
```

Comment:
What about setting `self.action_kwargs = None` in `__init__()` and then:

```suggestion
        if self.action_kwargs is None:
            raise NameError(
                '_prepare_kwargs() is required to be ran prior to _setup_connection()')
```

Reply:
Agreed. This is more explicit and cleaner.